### PR TITLE
Report error if lockfile generation fails

### PR DESCRIPTION
**Pull request by illicitonion**
Currently we ignore the exit code from this command, swallowing the
error, and typically fail in the subsequent call to `cargo` with a
message along the lines of "No Cargo.lock file was found".

Instead, fatally report the error from `cargo generate-lockfile`.

### CODE REVIEW

File: impl/src/metadata.rs [hunk 1]
```diff
@@ -91,11 +91,18 @@ impl LockfileGenerator for CargoLockfileGenerator {
     let lockfile_path = crate_root_dir.join("Cargo.lock");
 
     // Generate lockfile
-    std::process::Command::new(&self.cargo_bin_path)
+    let output = std::process::Command::new(&self.cargo_bin_path)
       .arg("generate-lockfile")
       .current_dir(&crate_root_dir)
       .output()?;
```
Comment (dfreese) [inline impl/src/metadata.rs:97]: LGTM, though, while you're here and working on error handling, would you mind adding a context to the potential error on this line?

**Review by dfreese — APPROVED** [review]

File: impl/src/metadata.rs (hunk 1 above)
Comment (illicitonion) [reply impl/src/metadata.rs]: Done - thanks!

**Review by dfreese — APPROVED** [review]
Thanks!

**Comment by illicitonion** [comment]
@dfreese Could I nudge you for a merge? Thanks!

**Comment by dfreese** [comment]
Sorry!  Approved before CI was ready and forgot about it.

**Pull request merged**